### PR TITLE
feat: update contributing docs

### PR DESCRIPTION
**Pull request by dpchamps**
**Description:**

Starting from a fresh pull, it can be difficult to get up and running. The changes I've proposed here represent findings from digging around the discord channel and a few closed issues.

I think it'd also be nice to call out a high-level explanation of what the difference between `svc_v1` and `svc_v2` is, but I don't know enough about the features. Happy to add it here if someone would give some guidence.

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -65,41 +65,59 @@ Steps to get started:
...
 
     ```bash
-    export RUST_BACKTRACE=full
+    export RUST_BACKTRACE=0
     export PATH="$PATH:$PWD/crates/swc_ecma_transforms/node_modules/.bin"
     export RUST_MIN_STACK=16777216
     ```
 
-4.  Install deno, if you are going to work on the bundler.
+4. Install deno
```
Comment (dpchamps) [inline CONTRIBUTING.md:92]: This, and `wasm32-wasi` is required for running all tests successfully

Comment (dpchamps) [inline CONTRIBUTING.md:92]: This, and `wasm32-wasi` is required for running all tests successfully

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -65,41 +65,59 @@ Steps to get started:
...
     yarn
     ( cd crates/swc_ecma_transforms; yarn )
     ```
 
-3.  Setup some environment variables which is required for tests.
+3. Setup some environment variables which is required for tests.
 
     ```bash
-    export RUST_BACKTRACE=full
+    export RUST_BACKTRACE=0
```
Comment (kdy1) [inline CONTRIBUTING.md:87]: This seems wrong?

File: CONTRIBUTING.md (hunk 1 above)
Comment (kdy1) [reply CONTRIBUTING.md:92]: But if you are not going to work on the bundler, you don't need to run bundler tests.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -65,41 +65,59 @@ Steps to get started:
...
     ```bash
     git submodule update --init --recursive
     ```
 
-2.  Install js dependencies.
+2. Install js dependencies.
 
     ```bash
     yarn
     ( cd crates/swc_ecma_transforms; yarn )
```
Comment (kwonoj) [inline CONTRIBUTING.md:81]: I think this is no longer required.

Comment (kdy1) [reply CONTRIBUTING.md:81]: Oh, you are right.

Comment (RiESAEX) [reply CONTRIBUTING.md:81]: I remember we need run yarn at root folder to install mocha,
and setup a path to node_modules/.bin

File: CONTRIBUTING.md (hunk 2 above)
Comment (dpchamps) [reply CONTRIBUTING.md:87]: see my description, perhaps theres something else that needs to be configured?

File: CONTRIBUTING.md (hunk 1 above)
Comment (dpchamps) [reply CONTRIBUTING.md:92]: As someone pulling the code for the first time, I usually want to run all tests to make sure everything went smoothly.

I tried to cover this by calling out "to run all tests..." in this subsection header.

Maybe a better way is to have a different subsection that outlines required dependencies per package, steps for running all tests, and a callout for how to run individual package tests?

File: CONTRIBUTING.md (hunk 3 above)
Comment (dpchamps) [reply CONTRIBUTING.md:81]: what should we change this to? no longer needed, or different command. 

Should also call out yarn as a dependency probably.

Comment (kdy1) [reply CONTRIBUTING.md:81]: No longer needed, but `yarn` in root directory is required.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -65,41 +65,59 @@ Steps to get started:
...
     ```
 
-3.  Setup some environment variables which is required for tests.
+3. Setup some environment variables which is required for tests.
 
     ```bash
-    export RUST_BACKTRACE=full
+    export RUST_BACKTRACE=0
     export PATH="$PATH:$PWD/crates/swc_ecma_transforms/node_modules/.bin"
```
Comment (RiESAEX) [inline CONTRIBUTING.md:88]: I think it should be `export PATH="$PATH:$PWD/node_modules/.bin"`

File: CONTRIBUTING.md (hunk 2 above)
Comment (kdy1) [reply CONTRIBUTING.md:87]: We need `RUST_LIB_BACKTRACE=0`

Comment (kdy1) [reply CONTRIBUTING.md:87]: We can configure it in the error reporting test, and I think we should.

File: CONTRIBUTING.md (hunk 1 above)
Comment (dpchamps) [reply CONTRIBUTING.md:92]: @kdy1 thoughts?

Comment (kdy1) [reply CONTRIBUTING.md:92]: Oh, yeah you are right.

File: CONTRIBUTING.md (hunk 2 above)
Comment (dpchamps) [reply CONTRIBUTING.md]: @kdy1 Great catch on this. `RUST_BACKTRACE=0` Was a suggestion I dug up on the discord server.

The error output is deceiving: it looked like it was emitting a backtrace... but actually, the captured error was being passed into `NormalizedOutput` with `Debug` formatting. 

With backtrace enabled, the lint errors were being captured with stack traces enabled. This is an easy fix: https://github.com/swc-project/swc/pull/3298/files#diff-d807373603a279f7b50a670e39c4b65357c79d13622bca50583de75692c3b6baL76

**Comment by dpchamps** [comment]
@RiESAEX , @kdy1 , @kwonoj  thanks for the feedback. I've updated based on your suggestions.

**Review by kdy1 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**